### PR TITLE
Add a canary test for KoreBuild

### PR DESCRIPTION
**Pull request by natemcmaster**
Uses local Korebuild changes and runs a full build of aspnet/DependencyInjection as a canary test.

@Eilon can we enable appveyor and travis for this repo?

### CODE REVIEW

File: test/test-korebuild.ps1 [hunk 1]
```diff
@@ -0,0 +1,11 @@
+$ErrorActionPreference ='Stop'
```
Comment (Eilon) [inline test/test-korebuild.ps1]: Maybe add a README.md to this folder with the basic description of what this is for? (Essentially your PR description?)

**Review by Eilon — CHANGES_REQUESTED** [review]
I'll add Travis/AppVeyor.

**Comment by Eilon** [comment]
Travis/AppVeyor enabled. The AppVeyor badge id is `9ik3b53tmisrdyif`. Can you also update the main README.md of this repo with CI status badges? (Copy from another repo, then change the AppVeyor badge id.)

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+init:
```
Comment (pranavkm) [inline appveyor.yml:1]: Can you set the `DOTNET_SKIP_FIRST_TIME_EXPERIENCE=true`, `DOTNET_CLI_TELEMETRY_OPTOUT=1` env variables in here?

**Review by pranavkm — APPROVED** [review]

File: appveyor.yml (hunk 2 above)
Comment (natemcmaster) [reply appveyor.yml:1]: Should we also set this in our appveyor files in the repos? We only set this in travis:

https://github.com/aspnet/DependencyInjection/blob/dev/appveyor.yml

Comment (pranavkm) [reply appveyor.yml:1]: Yes please. I think it does help speeding up our builds. Plus the expanding % thing at the top of our build logs looks kinda ridiculous

**Pull request merged**